pwa/src/components/paginationLimitSelect/PaginationLimitSelect.tsx

```diff
@@ -0,0 +1,72 @@
+import * as React from "react";
+import * as styles from "./PaginationLimitSelect.module.css";
+import clsx from "clsx";
+import { useForm } from "react-hook-form";
+import { SelectSingle } from "@conduction/components";
+import { IQueryLimitContext, queryLimitDefault, useQueryLimitContext } from "../../context/queryLimit";
+import { useTranslation } from "react-i18next";
+
+interface PaginationLimitSelectProps {
+  queryLimitName: string;
+  layoutClassName?: string;
+}
+
+export const PaginationLimitSelectComponent: React.FC<PaginationLimitSelectProps> = ({
+  queryLimitName,
+  layoutClassName,
+}) => {
+  const {
+    watch,
+    register,
+    control,
+    setValue,
+    formState: { errors },
+  } = useForm();
+  const { queryLimit, setQueryLimit } = useQueryLimitContext();
+  const { t } = useTranslation();
+
+  const watchLimit = watch("limit");
+
+  const value = queryLimit[queryLimitName as keyof IQueryLimitContext];
+
+  React.useEffect(() => {
+    if (!watchLimit) return;
+    if (parseInt(watchLimit.value) === value) return;
+
+    const selectedLimit = limitSelectOptions.find((LimitOption) => LimitOption.value === watchLimit.value);
+
+    selectedLimit !== undefined && setQueryLimit({ ...queryLimit, [queryLimitName]: parseInt(selectedLimit.value) });

```

Comment:
For better readability, I'd opt to make this a if statement, like so:

```
if (selectedLimit !== undefined) { ... } 
```

Maybe even `if (selectedLimit)` would suffice.